vcs/repository.go

```diff
@@ -175,3 +175,11 @@ type Tags []*Tag
 func (p Tags) Len() int           { return len(p) }
 func (p Tags) Less(i, j int) bool { return p[i].Name < p[j].Name }
 func (p Tags) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+// A FileLister is a repository that can perform actions related to
+// listing the entire file tree.
+type FileLister interface {
+	// ListFiles returns list of all file names in the repo at the
+	// given commit.
```

Comment:
My biggest concern is with the documentation here. It's extremely unhelpful in that it does not the answer the main question I wanted to see answered when I came here to read it.

Without actually running the command and looking at the output, after reading this, I have no hint as to whether the output is going to include elements like:

- `filename_only.txt` (just the name of file, not full path)
- `full/filepath.txt`
- `/absolute/full/filepath.txt` (rooted path, first character is `/`)
- `filepath\that\is\backslash\separated\on\non\linux\os.txt` (backslash separated path on non-unix OS)

Since go-vcs is meant to be a general abstraction on top of specific VCS systems, and also cross-platform, I think it's important to clarify whether the paths are filepath separator separated (meaning you will get `/` on Linux, OS X, others but `\` on Windows, etc.) or always forward-slash separated paths (so `/` on all OSes regardless of their filepath separator character).

I think it's better to always return consistent results, and go with returning forward-slash separated paths in all cases.

I've done a quick test, and it seems that `git` returns forward-slash separated paths even on Windows, so there's no need to use `filepath.FromSlash`.

So, my suggested changes to the docs would be:

1. Mention that paths are forward-slash-separated.
2. Mention that full paths (including directories, but not rooted) are returned.

A quick example would be very helpful:

> E.g., returned paths have the form "full/path/filename.txt".